### PR TITLE
Set max length 256 on names and add indexes

### PR DESCRIPTION
**Pull request by HaoK**
Fixes https://github.com/aspnet/Identity/issues/370

@divega does this look ok to you?

### CODE REVIEW

**Comment by HaoK** [comment]
I verified that the table definitions look like this:

CREATE TABLE [dbo].[AspNetUsers] (
    [Id]                   INT                IDENTITY (1, 1) NOT NULL,
    [AccessFailedCount]    INT                NOT NULL,
    [ConcurrencyStamp]     NVARCHAR (MAX)     NULL,
    [Email]                NVARCHAR (256)     NULL,
    [EmailConfirmed]       BIT                NOT NULL,
    [LockoutEnabled]       BIT                NOT NULL,
    [LockoutEnd]           DATETIMEOFFSET (7) NULL,
    [NormalizedEmail]      NVARCHAR (256)     NULL,
    [NormalizedUserName]   NVARCHAR (256)     NULL,
    [PasswordHash]         NVARCHAR (MAX)     NULL,
    [PhoneNumber]          NVARCHAR (MAX)     NULL,
    [PhoneNumberConfirmed] BIT                NOT NULL,
    [SecurityStamp]        NVARCHAR (MAX)     NULL,
    [TwoFactorEnabled]     BIT                NOT NULL,
    [UserName]             NVARCHAR (256)     NULL,
    CONSTRAINT [PK_IntUser] PRIMARY KEY CLUSTERED ([Id] ASC)
);
GO
CREATE NONCLUSTERED INDEX [IX_IntUser_NormalizedUserName_NormalizedEmail]
    ON [dbo].[AspNetUsers]([NormalizedUserName] ASC, [NormalizedEmail] ASC);

CREATE TABLE [dbo].[AspNetRoles] (
    [Id]               INT            IDENTITY (1, 1) NOT NULL,
    [ConcurrencyStamp] NVARCHAR (MAX) NULL,
    [Name]             NVARCHAR (256) NULL,
    [NormalizedName]   NVARCHAR (256) NULL,
    CONSTRAINT [PK_IntRole] PRIMARY KEY CLUSTERED ([Id] ASC)
);
GO
CREATE NONCLUSTERED INDEX [IX_IntRole_NormalizedName]
    ON [dbo].[AspNetRoles]([NormalizedName] ASC);

File: src/Microsoft.AspNet.Identity.EntityFramework/IdentityDbContext.cs [hunk 1]
```diff
@@ -33,9 +33,14 @@ protected override void OnModelCreating(ModelBuilder builder)
             builder.Entity<TUser>(b =>
                 {
                     b.Key(u => u.Id);
+                    b.Index("NormalizedUserName", "NormalizedEmail");
```
Comment (divega) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityDbContext.cs]: I think the two should be separate indexes unless we only have a query that searches for both together.

Comment (HaoK) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityDbContext.cs]: This is me misusing the API, I thought I was specifying 2 indexes, oops, I guess what I meant to do is to have two calls to Index...will update

Comment (HaoK) [inline src/Microsoft.AspNet.Identity.EntityFramework/IdentityDbContext.cs]: Updated and new table definition now correctly has two indexes

```
CREATE TABLE [dbo].[AspNetUsers] (
    [Id]                   INT                IDENTITY (1, 1) NOT NULL,
    [AccessFailedCount]    INT                NOT NULL,
    [ConcurrencyStamp]     NVARCHAR (MAX)     NULL,
    [Email]                NVARCHAR (256)     NULL,
    [EmailConfirmed]       BIT                NOT NULL,
    [LockoutEnabled]       BIT                NOT NULL,
    [LockoutEnd]           DATETIMEOFFSET (7) NULL,
    [NormalizedEmail]      NVARCHAR (256)     NULL,
    [NormalizedUserName]   NVARCHAR (256)     NULL,
    [PasswordHash]         NVARCHAR (MAX)     NULL,
    [PhoneNumber]          NVARCHAR (MAX)     NULL,
    [PhoneNumberConfirmed] BIT                NOT NULL,
    [SecurityStamp]        NVARCHAR (MAX)     NULL,
    [TwoFactorEnabled]     BIT                NOT NULL,
    [UserName]             NVARCHAR (256)     NULL,
    CONSTRAINT [PK_IntUser] PRIMARY KEY CLUSTERED ([Id] ASC)
);


GO
CREATE NONCLUSTERED INDEX [IX_IntUser_NormalizedEmail]
    ON [dbo].[AspNetUsers]([NormalizedEmail] ASC);


GO
CREATE NONCLUSTERED INDEX [IX_IntUser_NormalizedUserName]
    ON [dbo].[AspNetUsers]([NormalizedUserName] ASC);
```

**Comment by HaoK** [comment]
Updated to use u => u.NormalizedUserName syntax, also added verification of default schema for columns and indexes

**Comment by divega** [comment]
:shipit:

**Comment by HaoK** [comment]
d8b3cf238e3a106ebb3e08b1eb9f5661f997cdc0

**Pull request closed without merging**